cartographer/mapping_2d/pose_graph/constraint_builder.cc

```diff
@@ -70,6 +81,7 @@ void ConstraintBuilder::MaybeAddConstraint(
   if (sampler_.Pulse()) {
     common::MutexLocker locker(&mutex_);
     constraints_.emplace_back();
+    kQueueLengthMetric->Set(constraints_.size());
```

Comment:
Queue lengths are often difficult because it assumes that ConstraintBuilder is a singleton.

Alternatively, it's often easier to have two counters (computations started and finished), and compute their difference in Prometheus to get the queue length.

Reply:
A MapBuilder owns only one ConstraintBuilder, so the assumption is ok here (also for cloud-based mapping).
It would measure the length here directly and remember the two-counter solution for other cases (like scan matchers).